src/event.ts

```diff
@@ -1,34 +1,31 @@
 import type { Task } from './task'
-import type { BenchEvents } from './types'
+import type { BenchEvents, BenchEventsOptionalTask, BenchEventsWithError, BenchEventsWithTask } from './types'
 
-const createBenchEvent = (eventType: BenchEvents, target?: Task) => {
-  const event = new globalThis.Event(eventType)
-  if (target) {
-    Object.defineProperty(event, 'task', {
-      configurable: false,
-      enumerable: false,
-      value: target,
-      writable: false,
-    })
-  }
-  return event
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+interface BenchEvent<K extends BenchEvents=BenchEvents, M extends 'bench' | 'task' = 'bench'> extends globalThis.Event {
```

Comment:
do we need the interface? since we'll be exporting the class itself anyway

how come we can't just declare this type on the class?